Build fails on Fedora 35 Branched due to old catch2 version

Just upgraded to FC35 Branched and noticed Waybar build fails as `catch2-2.13.3` is not compatible with `glibc 2.34`:

```
[76/97] Compiling C++ object test/config_test.p/config.cpp.o
FAILED: test/config_test.p/config.cpp.o
ccache c++ -Itest/config_test.p -Itest -I../test -I../include -I../subprojects/Catch2-2.13.3/single_include -I../subprojects/fmt-7.1.3/include -I../subprojects/spdlog-1.8.5/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++17 -O3 '-DVERSION="0.9.8-50-g8534175 (branch '"'"'master'"'"')"' -DHAVE_LANGINFO_1STDAY '-DSYSCONFDIR="//usr/local/etc"' -DHAVE_CPU_LINUX -DHAVE_MEMORY_LINUX -DHAVE_SWAY -DHAVE_WLR -DHAVE_RIVER -DHAVE_LIBNL -DHAVE_LIBPULSE -DHAVE_DBUSMENU -DHAVE_LIBUDEV -DHAVE_LIBEVDEV -DHAVE_GTK_LAYER_SHELL -DHAVE_LIBDATE -pthread -DSPDLOG_FMT_EXTERNAL -MD -MQ test/config_test.p/config.cpp.o -MF test/config_test.p/config.cpp.o.d -o test/config_test.p/config.cpp.o -c ../test/config.cpp
In file included from /usr/include/signal.h:328,
                 from ../subprojects/Catch2-2.13.3/single_include/catch2/catch.hpp:8034,
                 from ../test/config.cpp:4:
../subprojects/Catch2-2.13.3/single_include/catch2/catch.hpp:10822:58: error: call to non-‘constexpr’ function ‘long int sysconf(int)’
10822 |     static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
      |                                                          ^~~~~~~~~~~
In file included from /usr/include/bits/sigstksz.h:24,
                 from /usr/include/signal.h:328,
                 from ../subprojects/Catch2-2.13.3/single_include/catch2/catch.hpp:8034,
                 from ../test/config.cpp:4:
/usr/include/unistd.h:641:17: note: ‘long int sysconf(int)’ declared here
  641 | extern long int sysconf (int __name) __THROW;
      |                 ^~~~~~~
In file included from ../test/config.cpp:4:
../subprojects/Catch2-2.13.3/single_include/catch2/catch.hpp:10881:45: error: size of array ‘altStackMem’ is not an integral constant-expression
10881 |     char FatalConditionHandler::altStackMem[sigStackSize] = {};
      |                                             ^~~~~~~~~~~~
```

This is trivially fixed by manually dropping the current version of catch2 into the `subprojects` folder and moving the relevant `meson.build` file in, but annoying as upstream hasn't been updated and is still on 2.13.3 rather than 2.13.7.

--- a/subprojects/catch2.wrap
+++ b/subprojects/catch2.wrap
@@ -1,11 +1,11 @@
 [wrap-file]
-directory = Catch2-2.13.3
-source_url = https://github.com/catchorg/Catch2/archive/v2.13.3.zip
-source_filename = Catch2-2.13.3.zip
-source_hash = 1804feb72bc15c0856b4a43aa586c661af9c3685a75973b6a8fc0b950c7cfd13
-patch_url = https://github.com/mesonbuild/catch2/releases/download/2.13.3-2/catch2.zip
-patch_filename = catch2-2.13.3-2-wrap.zip
-patch_hash = 21b590ab8c65b593ad5ee8f8e5b822bf9877b2c2672f97fbb52459751053eadf
+directory = Catch2-2.13.7
+source_url = https://github.com/catchorg/Catch2/archive/v2.13.7.zip
+source_filename = Catch2-2.13.7.zip
+source_hash = 3f3ccd90ad3a8fbb1beeb15e6db440ccdcbebe378dfd125d07a1f9a587a927e9
+patch_filename = catch2_2.13.7-1_patch.zip
+patch_url = https://wrapdb.mesonbuild.com/v2/catch2_2.13.7-1/get_patch
+patch_hash = 2f7369645d747e5bd866317ac1dd4c3d04dc97d3aad4fc6b864bdf75d3b57158
 
 [provide]
 catch2 = catch2_dep

